samples/ChangefeedMigrationSample/Program.cs

```diff
@@ -0,0 +1,222 @@
+            new RequestOptions { OfferThroughput = 400 });
+            await StartChangeFeedHost();
+        }
+
+        async Task StartChangeFeedHost()
+        {
+            string hostName = Guid.NewGuid().ToString();
+
+            // monitored collection info 
+            DocumentCollectionInfo documentCollectionLocation = new DocumentCollectionInfo
+            {
+                Uri = new Uri(this.uri),
+                MasterKey = this.secretKey,
+                DatabaseName = this.dbName,
+                CollectionName = this.monitoredCollectionName
+            };
+
+            // lease collection info 
+            DocumentCollectionInfo leaseCollectionLocation = new DocumentCollectionInfo
+            {
+                Uri = new Uri(this.uri),
+                MasterKey = this.secretKey,
+                DatabaseName = this.dbName,
+                CollectionName = this.leaseCollectionName
+            };
+
+            ChangeFeedEventHost host = new ChangeFeedEventHost(hostName, documentCollectionLocation, leaseCollectionLocation);
+            await host.RegisterObserverAsync<DocumentFeedObserver>();
+            Console.WriteLine("Main program: press Enter to stop...");
+            Console.ReadLine();
+            await host.UnregisterObserversAsync();
+        }
+
+        private async Task UpdateDb()
+        {
+            // Use this function to update data in monitored collection in seperate thread
+            // Returns all documents in the collection.
+            Console.WriteLine("Connect client");
+            DocumentClient client = new DocumentClient(new Uri(this.uri), this.secretKey);
+            Uri collectionUri = UriFactory.CreateDocumentCollectionUri(this.dbName, this.monitoredCollectionName);
+
+            Console.WriteLine("Connect database");
+            try
+            {
+                Database database = await client.ReadDatabaseAsync(UriFactory.CreateDatabaseUri(this.dbName));
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine("Connect database failed");
+                Console.WriteLine(e.Message);
+            }
+
+            // Create new documents 
+            System.Console.WriteLine("Create JSON document");
+            for (int i = 0; i < 10; i++)
+            {
+                System.Console.WriteLine("Creating document XMS-0004 {0}", i);
+                await client.UpsertDocumentAsync(
+                UriFactory.CreateDocumentCollectionUri(this.dbName, this.monitoredCollectionName),
+                new DeviceReading
+                {
+                    Id = String.Join("XMS-005-FE24C_", i.ToString()),
+                    DeviceId = "XMS-0005",
+                    MetricType = "Temperature",
+                    MetricValue = 80.00 + (float)i,
+                    Unit = "Fahrenheit",
+                    ReadingTime = DateTime.UtcNow
+                });
+                System.Threading.Thread.Sleep(5000);
```

Comment:
Don't do that from async method. Use Task.Wait. Also use something like TimeSpan.FromSeconds(5) rather than int which is more clear.